rovers_parte4_dominio.pddl

```diff
@@ -0,0 +1,206 @@
+              )
+              :effect (and (have_image ?r ?o ?m)
+                     (not (calibrated ?i ?r))
+              )
+       )
+
+       ;comunica data del suelo (necesita haber agarrado suelo)
+       (:action communicate_soil_data
+              :parameters (?r - rover ?l - lander ?p - waypoint ?x - waypoint ?y - waypoint)
+              :precondition (and (at ?r ?x)
+                     (at_lander ?l ?y)(have_soil_analysis ?r ?p)
+                     (visible ?x ?y)(available ?r)(channel_free ?l)
+              )
+              :effect (and (not (available ?r))
+                     (not (channel_free ?l))(channel_free ?l)
+                     (communicated_soil_data ?p)(available ?r)
+              )
+       )
+
+       ;comunica data de roca (necesita haber agarrado roca)
+       (:action communicate_rock_data
+              :parameters (?r - rover ?l - lander ?p - waypoint ?x - waypoint ?y - waypoint)
+              :precondition (and (at ?r ?x)
+                     (at_lander ?l ?y)(have_rock_analysis ?r ?p)
+                     (visible ?x ?y)(available ?r)(channel_free ?l)
+              )
+              :effect (and (not (available ?r))
+                     (not (channel_free ?l))(channel_free ?l)(communicated_rock_data ?p)(available ?r)
+              )
+       )
+
+       ;comunica data de imagen (necesita haber tomado foto)
+       (:action communicate_image_data
+              :parameters (?r - rover ?l - lander ?o - objective ?m - mode ?x - waypoint ?y - waypoint)
+              :precondition (and (at ?r ?x)
+                     (at_lander ?l ?y)(have_image ?r ?o ?m)(visible ?x ?y)(available ?r)(channel_free ?l)
+              )
+              :effect (and (not (available ?r))
+                     (not (channel_free ?l))(channel_free ?l)(communicated_image_data ?o ?m)(available ?r)
+              )
+       )
+
+       ;Esta bateria se va agotando cada vez que se hace una accion de movimientos - niega battery level para setear nuevo battery level (descargar)
+       (:action discharge_battery_level
+              :parameters (?r - rover ?b - battery ?v - batterylevel)
+              :precondition (and (at ?r ?x)
+                     (at_lander ?l ?y)(have_image ?r ?o ?m)(visible ?x ?y)(available ?r)(channel_free ?l)
+              )
+              :effect (and (not ((at_charge ?b - battery ?v - batterylevel)))
+                    ((at_charge ?b - battery ?t - batterylevel))
+              )
+       )
+
+       ;valida rover en lander y niega battery level para setear nuevo battery level
+       (:action charge_battery
+              :parameters (?r - rover ?x - waypoint ?l - lander ?b - battery ?v - batterylevel ?t - batterylevel)
+              :precondition (and (at ?r ?x) (at_lander ?l ?x)(available ?r)
+              )
+              :effect (and (not ((at_charge ?b - battery ?v - batterylevel)))
+                    ((at_charge ?b - battery ?t - batterylevel))
+              )
+       )
```

Comment:
checks if rover is at lander and the effect it's the battery from any value to highest value possible